php-checks/src/main/java/org/sonar/php/checks/phpunit/AssertionArgumentOrderCheck.java

```diff
@@ -52,23 +57,47 @@ public void visitFunctionCall(FunctionCallTree tree) {
     if (arguments.size() >= 2 && assertion.isPresent() && assertion.get().hasExpectedValue() && !hasNamedArgument(tree)) {
       ExpressionTree expected = arguments.get(0).value();
       ExpressionTree actual = arguments.get(1).value();
-      if (isLiteralOrClassName(actual) && !isLiteralOrClassName(expected)) {
+      if (isLiteralOrClassNameOrParameter(actual) && !isLiteralOrClassNameOrParameter(expected)) {
         newIssue(actual, MESSAGE).secondary(expected, SECONDARY_MESSAGE);
       }
     }
 
     super.visitFunctionCall(tree);
   }
 
-  private static boolean isLiteralOrClassName(ExpressionTree expression) {
+  private static boolean isLiteralOrClassNameOrParameter(ExpressionTree expression) {
     return assignedValue(expression).is(LITERAL) ||
-      (expression instanceof MemberAccessTree &&
-        isStaticAccessWithName((MemberAccessTree) expression, "class"));
+      isStaticAccessWithName(expression, "class") ||
+      isDefinedAsParameter(expression);
   }
 
-  private static boolean isStaticAccessWithName(MemberAccessTree tree, String memberName) {
-    return tree.isStatic() &&
-      tree.member().is(Kind.NAME_IDENTIFIER) &&
-      memberName.equals(((NameIdentifierTree) tree.member()).text());
+  private static boolean isStaticAccessWithName(ExpressionTree expression, String memberName) {
+    if (expression instanceof MemberAccessTree) {
+      MemberAccessTree tree = (MemberAccessTree) expression;
+      return tree.isStatic() && memberName.equals(name(tree.member()));
+    }
+    return false;
+  }
+
+  private static boolean isDefinedAsParameter(ExpressionTree expression) {
+    MethodDeclarationTree method = (MethodDeclarationTree) TreeUtils.findAncestorWithKind(expression, Kind.METHOD_DECLARATION);
+    if (method != null) {
+      String name = name(expression);
+      for (ParameterTree parameter : method.parameters().parameters()) {
+        String text = parameter.variableIdentifier().text();
+        if (text.equals(name)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  @CheckForNull
+  private static String name(Tree expression) {
```

Comment:
Do we have any common locations for such utilities? I feel like they'll get scattered across code base

Reply:
I thought about extending `CheckUtils::nameOf()`, but the potential impact is big IMO. So I decided to make it local.